firestore/tests/unit/v1/test_client.py

```diff
@@ -63,6 +63,22 @@ def test_constructor_explicit(self):
         self.assertEqual(client._database, database)
         self.assertIs(client._client_info, client_info)
 
+    def test_constructor_with_emulator_host(self):
+        from google.cloud.environment_vars import FIRESTORE_EMULATOR
+
+        credentials = _make_credentials()
+        emulator_host = "localhost:8081"
+        with mock.patch("os.getenv") as getenv:
+            getenv.return_value = emulator_host
+            with mock.patch("grpc.insecure_channel") as factory:
+                getenv.return_value = emulator_host
+                client = self._make_one(project=self.PROJECT, credentials=credentials)
+
+        self.assertEqual(client._emulator_host, emulator_host)
+        self.assertIs(client._emulator_channel, factory.return_value)
+        factory.assert_called_once_with(emulator_host)
```

Comment:
The lines 78, 79 seem to be outside of the `factory` context defined in line 73.

Reply:
yes, but it work fine, if you want i''ll move it